### PR TITLE
Refactor metal3-dev-env to accomodate BMO changes

### PR DESCRIPTION
**Pull request by kashifest**
This PR refactors metal3-dev-env to accomodate BMO changes.

### CODE REVIEW

File: 03_launch_mgmt_cluster.sh [hunk 1]
```diff
@@ -378,15 +378,13 @@ function update_capm3_imports(){
...
-  BMO_REAL_PATH="$(realpath --relative-to="${CAPM3PATH}/config/bmo" "${BMOPATH}")"
-  for folder in $FOLDERS; do
-    sed -i -e "s#github.com/metal3-io/baremetal-operator/${folder}?ref=.*#${BMO_REAL_PATH}/${folder}#" "config/bmo/kustomization.yaml"
-  done
+  cp config/ipam/kustomization.yaml config/ipam/kustomization.yaml.orig
 
-  # Render the IPAM components from local repo instead of using the released version
   make hack/tools/bin/kustomize
+  ./hack/tools/bin/kustomize build "${BMOPATH}/config/default" > config/bmo/bmo-components.yaml
+  sed -i -e "s#https://raw.githubusercontent.com/metal3-io/baremetal-operator/master/config/render/capm3.yaml#config/bmo/bmo-components.yaml#" "config/bmo/kustomization.yaml"
```
Comment (maelk) [inline 03_launch_mgmt_cluster.sh:385]:
```suggestion
  sed -i -e "s#https://raw.githubusercontent.com/metal3-io/baremetal-operator/master/config/render/capm3.yaml#bmo-components.yaml#" "config/bmo/kustomization.yaml"
```

Comment (kashifest) [reply 03_launch_mgmt_cluster.sh]: Done

**Review by maelk — APPROVED** [review]

File: 03_launch_mgmt_cluster.sh [hunk 2]
```diff
@@ -230,8 +230,8 @@ EOF
   "${BMOPATH}/tools/deploy.sh" true false "${IRONIC_TLS_SETUP}" "${IRONIC_BASIC_AUTH}" true
 
   # Restore original files
-  mv "${BMOPATH}/deploy/default/ironic_bmo_configmap.env.orig" "${BMOPATH}/deploy/default/ironic_bmo_configmap.env"
-  mv "${BMOPATH}/deploy/operator/bmo.yaml.orig" "${BMOPATH}/deploy/operator/bmo.yaml"
+  mv "${BMOPATH}/config/default/ironic.env.orig" "${BMOPATH}/config/default/ironic.env"
+  mv "${BMOPATH}/config/manager/manager.yaml.orig" "${BMOPATH}/config/manager/manager.yaml.orig"
```
Comment (maelk) [inline 03_launch_mgmt_cluster.sh:234]: the second one should not have .orig

File: 04_verify.sh [hunk 3]
```diff
@@ -204,7 +204,7 @@ EXPTD_V1ALPHA3_RS="cluster.x-k8s.io/provider:infrastructure-metal3:capm3-system:
   cluster.x-k8s.io/provider:cluster-api:capi-webhook-system:1 \
   cluster.x-k8s.io/provider:bootstrap-kubeadm:capi-webhook-system:1 \
   cluster.x-k8s.io/provider:control-plane-kubeadm:capi-webhook-system:1 \
-  name:metal3-baremetal-operator:metal3:1"
+  name:baremetal-operator-controller-manager:metal3:1"
```
Comment (maelk) [inline 04_verify.sh:207]: isn't this in the wrong namespace ?

Comment (kashifest) [reply 04_verify.sh:207]: yes that is

File: 03_launch_mgmt_cluster.sh [hunk 4]
```diff
@@ -256,7 +256,7 @@ EOF
 
     touch bmo.out.log
     touch bmo.err.log
-    kubectl scale deployment metal3-baremetal-operator -n metal3 --replicas=0
+    kubectl scale deployment baremetal-operator-controller-manager -n metal3 --replicas=0
```
Comment (maelk) [inline 03_launch_mgmt_cluster.sh:259]: the namespace should be baremetal-operator-system

Comment (maelk) [reply 03_launch_mgmt_cluster.sh:259]: we can fix that later, to get the CI back running

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

File: 03_launch_mgmt_cluster.sh (hunk 4 above)
Comment (furkatgofurov7) [reply 03_launch_mgmt_cluster.sh:259]: I will put a patch to fix the namespace

**Pull request merged**